### DISCUSSION TITLE
Improve data ingestion on Azure demo with unstructured

### DISCUSSION THREAD

**Discussion by nhtkid** [Unstructured-IO/unstructured#754] [Q&A]
Hi,

Can anyone help on fork this https://github.com/ajananth/openai-cogsrch-fsi to improve the data ingestion.
Right now, this sample app only supports PDF files.
I am thinking by using unstructured library, we should be able to support all kinds of file types.

Thanks.

**Comment by MthwRobinson** [comment]
@nhtkid - Thanks for the post. @ryannikolaidis has been doing a lot of work on the ingestion side of things and may be able to provide some suggestions.

**Comment by cragwolfe** [comment]
@nhtkid , yes, you could use unstructured to process a large variety of file types. There also exists an azure blob storage connector which can process all the documents in a azure blob storage container and run them through unstructured's `partition` function. I'm not sure that'd be useful for your project but in case it is please see https://github.com/Unstructured-IO/unstructured/blob/main/test_unstructured_ingest/test-ingest-azure.sh for example usage.
